task_sdk/src/airflow/sdk/log.py

```diff
@@ -0,0 +1,371 @@
+    return event_dict
+
+
+class StdBinaryStreamHandler(logging.StreamHandler):
+    """A logging.StreamHandler that sends logs as binary JSON over the given stream."""
+
+    stream: BinaryIO
+
+    def __init__(self, stream: BinaryIO):
+        super().__init__(stream)
+
+    def emit(self, record: logging.LogRecord):
+        try:
+            msg = self.format(record)
+            buffer = bytearray(msg, "ascii", "backslashreplace")
+
+            buffer += b"\n"
+
+            stream = self.stream
+            stream.write(buffer)
+            self.flush()
+        except RecursionError:  # See issue 36272
+            raise
+        except Exception:
+            self.handleError(record)
+
+
+@cache
+def logging_processors(
+    enable_pretty_log: bool,
+):
+    if enable_pretty_log:
+        timestamper = structlog.processors.MaybeTimeStamper(fmt="%Y-%m-%d %H:%M:%S.%f")
+    else:
+        timestamper = structlog.processors.MaybeTimeStamper(fmt="iso")
+
+    processors: list[structlog.typing.Processor] = [
+        timestamper,
+        structlog.contextvars.merge_contextvars,
+        structlog.processors.add_log_level,
+        structlog.stdlib.PositionalArgumentsFormatter(),
+        logger_name,
+        redact_jwt,
+        structlog.processors.StackInfoRenderer(),
+    ]
+
+    if enable_pretty_log:
+        # Imports to suppress showing code from these modules
+        import asyncio
+        import contextlib
+
+        import click
+        import httpcore
+        import httpx
+        import typer
+
+        rich_exc_formatter = structlog.dev.RichTracebackFormatter(
+            # These values are picked somewhat arbitrarily to produce useful-but-compact tracebacks. If
+            # we ever need to change these then they should be configurable.
+            extra_lines=0,
+            max_frames=30,
+            indent_guides=False,
+            suppress=[asyncio, httpcore, httpx, contextlib, click, typer],
+        )
+        my_styles = structlog.dev.ConsoleRenderer.get_default_level_styles()
+        my_styles["debug"] = structlog.dev.CYAN
+
+        console = structlog.dev.ConsoleRenderer(
+            exception_formatter=rich_exc_formatter, level_styles=my_styles
+        )
+        processors.append(console)
+        return processors, {
+            "timestamper": timestamper,
+            "console": console,
+        }
+    else:
+        # Imports to suppress showing code from these modules
+        import asyncio
+        import contextlib
+
+        import click
+        import httpcore
+        import httpx
+        import typer
+
+        dict_exc_formatter = structlog.tracebacks.ExceptionDictTransformer(
+            use_rich=False, show_locals=False, suppress=(click, typer)
+        )
+
+        dict_tracebacks = structlog.processors.ExceptionRenderer(
+            structlog.tracebacks.ExceptionDictTransformer(
+                use_rich=False, show_locals=False, suppress=(click, typer)
```

Comment:
Do we need to add `asyncio, httpcore, httpx, contextlib` too or remove the imports in the `else` clause?

Reply:
Yeah updated.